src/twitterNameResolver.ts

```diff
@@ -1,17 +1,32 @@
 import {fetchSafe} from "./utils";
 
+const callTwitterApi = async (url: string, BEARER_TOKEN: string) => {
+    const response = await fetchSafe(url, {
+        method: 'GET',
+        headers: {
+            'Authorization': `Bearer ${BEARER_TOKEN}`,
+        },
+    });
+    if (!response.ok) {
+        throw new Error(`Error: ${response.status}`);
+    }
+    return await response.json();
+}
+
 export class TwitterNameResolver {
+    protected TWITTER_BEARER_TOKEN = "";
+
+    constructor(twitterApiKey: string) {
+        this.TWITTER_BEARER_TOKEN = twitterApiKey;
+    }
+
     async getTwitterID(inputCombination: string): Promise<string> {
-        const response = await fetchSafe("https://www.idriss.xyz/v1/getTwitterID?identifier=" + encodeURIComponent(inputCombination));
-        if (response.status != 200) throw new Error("IDriss api responded with code " + response.status + " " + response.statusText + "\r\n" + await response.text())
-        const json = await response.json();
-        return json.twitterID;
+        const json = await callTwitterApi(`https://api.twitter.com/2/users/by/username/${inputCombination}`, this.TWITTER_BEARER_TOKEN);
+        return json?.data?.id;
     }
 
     async reverseTwitterID(id: string): Promise<string> {
-        const response = await fetchSafe("https://www.idriss.xyz/v1/getTwitterNames?ids=" + encodeURIComponent(id));
-        if (response.status != 200) throw new Error("IDriss api responded with code " + response.status + " " + response.statusText + "\r\n" + await response.text())
-        const json = await response.json();
-        return json.twitterNames[id];
+        const json = await callTwitterApi(`https://api.twitter.com/2/users/${id}`, this.TWITTER_BEARER_TOKEN);
+        return json?.data?.username;
     }
```

Comment:
Since this is some 3rd party api I think we should improve it by validating twitter response against some schema, WDYT?